packages/flutter/lib/src/material/dialog.dart

```diff
@@ -117,22 +168,104 @@ class Dialog extends StatelessWidget {
       ));
     }
 
-    return new Center(
-      child: new Container(
-        margin: new EdgeInsets.symmetric(horizontal: 40.0, vertical: 24.0),
+    return new Dialog(
+      child: new IntrinsicWidth(
+        child: new Column(
+          mainAxisSize: MainAxisSize.min,
+          crossAxisAlignment: CrossAxisAlignment.stretch,
+          children: children
```

Comment:
what if the children are too high for the window? surely we still need a scrolling widget here.